redis/asyncio/sentinel.py

```diff
@@ -141,31 +208,70 @@ def owns_connection(self, connection: Connection):
         return check and super().owns_connection(connection)
 
     async def get_master_address(self):
-        master_address = await self.sentinel_manager.discover_master(self.service_name)
-        if self.is_master:
-            if self.master_address != master_address:
-                self.master_address = master_address
-                # disconnect any idle connections so that they reconnect
-                # to the new master the next time that they are used.
-                await self.disconnect(inuse_connections=False)
-        return master_address
+        return await self.proxy.get_master_address()
 
-    async def rotate_slaves(self) -> AsyncIterator:
+    def rotate_slaves(self) -> AsyncIterator:
         """Round-robin slave balancer"""
-        slaves = await self.sentinel_manager.discover_slaves(self.service_name)
-        if slaves:
-            if self.slave_rr_counter is None:
-                self.slave_rr_counter = random.randint(0, len(slaves) - 1)
-            for _ in range(len(slaves)):
-                self.slave_rr_counter = (self.slave_rr_counter + 1) % len(slaves)
-                slave = slaves[self.slave_rr_counter]
-                yield slave
-        # Fallback to the master connection
-        try:
-            yield await self.get_master_address()
-        except MasterNotFoundError:
-            pass
-        raise SlaveNotFoundError(f"No slave found for {self.service_name!r}")
+        return self.proxy.rotate_slaves()
+
+
+class SentinelBlockingConnectionPool(BlockingConnectionPool):
```

Comment:
For feature parity it needs to be added it for sync API as well

Reply:
The synchronous version of connection pool has issues with LSP that currently prevent this from being done:
1. https://github.com/redis/redis-py/blob/master/redis/connection.py#L1253
2. https://github.com/redis/redis-py/blob/master/redis/connection.py#L1459
3. https://github.com/redis/redis-py/blob/master/redis/sentinel.py#L117
Solving this problem will require a complete refactoring of the synchronous connection pool

Reply:
Thank you for pointing out! I'll create an issue to keep track of this. Let's skip sync implementation for now